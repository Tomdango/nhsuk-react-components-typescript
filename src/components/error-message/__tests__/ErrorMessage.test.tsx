import React from 'react';
import ErrorMessage from '..';
import { shallow } from 'enzyme';

describe('ErrorMessage', () => {
  it('matches snapshot', () => {
    const element = shallow(<ErrorMessage>Error</ErrorMessage>);
    expect(element).toMatchSnapshot('ErrorMessage');
    element.unmount();
  });

  it('has default visuallyHiddenText', () => {
    const element = shallow(<ErrorMessage>Error</ErrorMessage>);
    expect(element.find('.nhsuk-u-visually-hidden').text()).toBe('Error: ');
    element.unmount();
  });

  it('has disabled visuallyHiddenText', () => {
    const element = shallow(<ErrorMessage visuallyHiddenText={false}>Error</ErrorMessage>);
    expect(element.find('.nhsuk-u-visually-hidden').exists()).toBeFalsy();
    element.unmount();
  });

  it('has custom visuallyHiddenText', () => {
    const element = shallow(<ErrorMessage visuallyHiddenText="Custom">Error</ErrorMessage>);
    expect(element.find('.nhsuk-u-visually-hidden').text()).toBe('Custom');
    element.unmount();
  });
});
