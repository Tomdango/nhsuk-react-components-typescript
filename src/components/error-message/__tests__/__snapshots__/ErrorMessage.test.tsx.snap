// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot: ErrorMessage 1`] = `
<span
  className="nhsuk-error-message"
>
  <span
    className="nhsuk-u-visually-hidden"
  >
    Error: 
  </span>
  Error
</span>
`;
